# ANALYZING MIMICKED HEALTHCARE DATASET USING SQL

## Dataset Information:
Each column provides specific information about the patient, their admission, and the healthcare services provided, making this dataset suitable for various data analysis and modeling tasks in the healthcare domain. Here's a brief explanation of each column in the dataset -

Name: This column represents the name of the patient associated with the healthcare record.

Age: The age of the patient at the time of admission, expressed in years.

Gender: Indicates the gender of the patient, either "Male" or "Female."

Blood Type: The patient's blood type, which can be one of the common blood types (e.g., "A+", "O-", etc.).

Medical Condition: This column specifies the primary medical condition or diagnosis associated with the patient, such as "Diabetes," "Hypertension," "Asthma," and more.

Date of Admission: The date the patient was admitted to the healthcare facility.

Doctor: The name of the doctor responsible for the patient's care during their admission.

Hospital: Identifies the healthcare facility or hospital where the patient was admitted.

Insurance Provider: This column indicates the patient's insurance provider, which can be one of several options, including "Aetna," "Blue Cross," "Cigna," "UnitedHealthcare," and "Medicare."

Billing Amount: The money billed for the patient's healthcare services during admission. This is expressed as a floating-point number.

Room Number: The room number where the patient was accommodated during their admission.

Admission Type: Specifies the type of admission, which can be "Emergency," "Elective," or "Urgent," reflecting the circumstances of the admission.

Discharge Date: The date on which the patient was discharged from the healthcare facility, based on the admission date and a random number of days within a realistic range.

Medication: Identifies a medication prescribed or administered to the patient during their admission. Examples include "Aspirin," "Ibuprofen," "Penicillin," "Paracetamol," and "Lipitor."

Test Results: Describes the results of a medical test conducted during the patient's admission. Possible values include "Normal," "Abnormal," or "Inconclusive," indicating the outcome of the test.


In [1]:
load_ext sql

In [113]:
%%sql
# Listing the number of tables in the database
SHOW TABLES;

 * mysql+pymysql://root:***@localhost:3306/healthcare
1 rows affected.


[('healthcare_dataset',)]

In [114]:
%%sql
# Let's check the type of data in the table
SELECT *
FROM healthcare_dataset
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/healthcare
5 rows affected.


[('Bobby JacksOn', 30, 'Male', 'B-', 'Cancer', '2024-01-31', 'Matthew Smith', 'Sons and Miller', 'Blue Cross', 18856.281305978155, 328, 'Urgent', '2024-02-02', 'Paracetamol', 'Normal'),
 ('LesLie TErRy', 62, 'Male', 'A+', 'Obesity', '2019-08-20', 'Samantha Davies', 'Kim Inc', 'Medicare', 33643.327286577885, 265, 'Emergency', '2019-08-26', 'Ibuprofen', 'Inconclusive'),
 ('DaNnY sMitH', 76, 'Female', 'A-', 'Obesity', '2022-09-22', 'Tiffany Mitchell', 'Cook PLC', 'Aetna', 27955.096078842456, 205, 'Emergency', '2022-10-07', 'Aspirin', 'Normal'),
 ('andrEw waTtS', 28, 'Female', 'O+', 'Diabetes', '2020-11-18', 'Kevin Wells', 'Hernandez Rogers and Vang,', 'Medicare', 37909.78240987528, 450, 'Elective', '2020-12-18', 'Ibuprofen', 'Abnormal'),
 ('adrIENNE bEll', 43, 'Female', 'AB+', 'Cancer', '2022-09-19', 'Kathleen Hanna', 'White-White', 'Aetna', 14238.317813937623, 458, 'Urgent', '2022-10-09', 'Penicillin', 'Abnormal')]

# Exploratory Data Analysis (EDA)

In [115]:
%%sql
# Checking structure of the data 

DESCRIBE healthcare_dataset;

 * mysql+pymysql://root:***@localhost:3306/healthcare
15 rows affected.


[('Name', 'text', 'YES', '', None, ''),
 ('Age', 'int', 'YES', '', None, ''),
 ('Gender', 'text', 'YES', '', None, ''),
 ('Blood Type', 'text', 'YES', '', None, ''),
 ('Medical Condition', 'text', 'YES', '', None, ''),
 ('Date of Admission', 'text', 'YES', '', None, ''),
 ('Doctor', 'text', 'YES', '', None, ''),
 ('Hospital', 'text', 'YES', '', None, ''),
 ('Insurance Provider', 'text', 'YES', '', None, ''),
 ('Billing Amount', 'double', 'YES', '', None, ''),
 ('Room Number', 'int', 'YES', '', None, ''),
 ('Admission Type', 'text', 'YES', '', None, ''),
 ('Discharge Date', 'text', 'YES', '', None, ''),
 ('Medication', 'text', 'YES', '', None, ''),
 ('Test Results', 'text', 'YES', '', None, '')]

### Changing datatypes of some features
1. Date of Admission" and "Discharge Date" from 'TEXT' format to 'DATETIME' format.
2. "Billing Amount" from Double to 'Numeric' (Not a must).

In [116]:
%%sql
ALTER TABLE healthcare_dataset
MODIFY COLUMN `Date of Admission` DATETIME,
MODIFY COLUMN `Discharge Date` DATETIME,
MODIFY COLUMN `Billing Amount` NUMERIC(15,2);

 * mysql+pymysql://root:***@localhost:3306/healthcare


55500 rows affected.


[]

### Renaming COLUMN  tittles

In [117]:
%%sql
ALTER TABLE healthcare_dataset
RENAME COLUMN Name TO name,
RENAME COLUMN Age TO age,
RENAME COLUMN Gender TO gender,
RENAME COLUMN `Blood Type` TO blood_type,
RENAME COLUMN `Medical Condition` TO medical_condition,
RENAME COLUMN `Date of Admission` TO date_of_admission,
RENAME COLUMN Doctor TO doctor,
RENAME COLUMN Hospital TO hospital,
RENAME COLUMN `Insurance Provider` TO insurance_provider,
RENAME COLUMN `Billing Amount` TO billing_amount,
RENAME COLUMN `Room Number` TO room_number,
RENAME COLUMN `Admission Type` TO admission_type,
RENAME COLUMN `Discharge Date` TO discharge_date,
RENAME COLUMN Medication TO medication,
RENAME COLUMN `Test Results` TO test_results;

 * mysql+pymysql://root:***@localhost:3306/healthcare
0 rows affected.


[]

In [118]:
%%sql
# Checking if the updates made have been effected

DESCRIBE healthcare_dataset;

 * mysql+pymysql://root:***@localhost:3306/healthcare
15 rows affected.


[('name', 'text', 'YES', '', None, ''),
 ('age', 'int', 'YES', '', None, ''),
 ('gender', 'text', 'YES', '', None, ''),
 ('blood_type', 'text', 'YES', '', None, ''),
 ('medical_condition', 'text', 'YES', '', None, ''),
 ('date_of_admission', 'datetime', 'YES', '', None, ''),
 ('doctor', 'text', 'YES', '', None, ''),
 ('hospital', 'text', 'YES', '', None, ''),
 ('insurance_provider', 'text', 'YES', '', None, ''),
 ('billing_amount', 'decimal(15,2)', 'YES', '', None, ''),
 ('room_number', 'int', 'YES', '', None, ''),
 ('admission_type', 'text', 'YES', '', None, ''),
 ('discharge_date', 'datetime', 'YES', '', None, ''),
 ('medication', 'text', 'YES', '', None, ''),
 ('test_results', 'text', 'YES', '', None, '')]

In [119]:
%%sql

SELECT *
FROM healthcare_dataset
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/healthcare
5 rows affected.


[('Bobby JacksOn', 30, 'Male', 'B-', 'Cancer', datetime.datetime(2024, 1, 31, 0, 0), 'Matthew Smith', 'Sons and Miller', 'Blue Cross', Decimal('18856.28'), 328, 'Urgent', datetime.datetime(2024, 2, 2, 0, 0), 'Paracetamol', 'Normal'),
 ('LesLie TErRy', 62, 'Male', 'A+', 'Obesity', datetime.datetime(2019, 8, 20, 0, 0), 'Samantha Davies', 'Kim Inc', 'Medicare', Decimal('33643.33'), 265, 'Emergency', datetime.datetime(2019, 8, 26, 0, 0), 'Ibuprofen', 'Inconclusive'),
 ('DaNnY sMitH', 76, 'Female', 'A-', 'Obesity', datetime.datetime(2022, 9, 22, 0, 0), 'Tiffany Mitchell', 'Cook PLC', 'Aetna', Decimal('27955.10'), 205, 'Emergency', datetime.datetime(2022, 10, 7, 0, 0), 'Aspirin', 'Normal'),
 ('andrEw waTtS', 28, 'Female', 'O+', 'Diabetes', datetime.datetime(2020, 11, 18, 0, 0), 'Kevin Wells', 'Hernandez Rogers and Vang,', 'Medicare', Decimal('37909.78'), 450, 'Elective', datetime.datetime(2020, 12, 18, 0, 0), 'Ibuprofen', 'Abnormal'),
 ('adrIENNE bEll', 43, 'Female', 'AB+', 'Cancer', datetime.datetime(2022, 9, 19, 0, 0), 'Kathleen Hanna', 'White-White', 'Aetna', Decimal('14238.32'), 458, 'Urgent', datetime.datetime(2022, 10, 9, 0, 0), 'Penicillin', 'Abnormal')]

In [320]:
%%sql

SELECT 
    COUNT(DISTINCT name) as number_unique_patient 
FROM healthcare_dataset;

 * mysql+pymysql://root:***@localhost:3306/healthcare
1 rows affected.


[(40235,)]

## Checking and handling duplicates 

Create a CTE called 'duplicates', and then check duplicated rows in the table

In [325]:
%%sql

WITH duplicates as (SELECT *, 
ROW_NUMBER() OVER (
    PARTITION BY name, age, doctor
    ORDER BY name
) as row_num
FROM healthcare_dataset)
(SELECT COUNT(*) as duplicated_rows FROM duplicates
    WHERE row_num >1);

 * mysql+pymysql://root:***@localhost:3306/healthcare
1 rows affected.


[(534,)]

From the above, we can deduce that there are 534 duplicates in the table. 

Why the large number of duplicates? Is there need to delete them?

In [122]:
%%sql
# Let's confirm indeed they are duplicates

SELECT * 
FROM healthcare_dataset
WHERE name = "ABIgaIL YOung"
AND age = 41;

 * mysql+pymysql://root:***@localhost:3306/healthcare
2 rows affected.


[('ABIgaIL YOung', 41, 'Female', 'O+', 'Hypertension', datetime.datetime(2022, 12, 15, 0, 0), 'Edward Kramer', 'Moore-Mcdaniel', 'UnitedHealthcare', Decimal('1983.57'), 192, 'Elective', datetime.datetime(2023, 1, 13, 0, 0), 'Ibuprofen', 'Normal'),
 ('ABIgaIL YOung', 41, 'Female', 'O+', 'Hypertension', datetime.datetime(2022, 12, 15, 0, 0), 'Edward Kramer', 'Moore-Mcdaniel', 'UnitedHealthcare', Decimal('1983.57'), 192, 'Elective', datetime.datetime(2023, 1, 13, 0, 0), 'Ibuprofen', 'Normal')]

In [123]:
%%sql

SELECT * 
FROM healthcare_dataset
WHERE name = "ANdrE WoOds"
AND age = 78;

 * mysql+pymysql://root:***@localhost:3306/healthcare
2 rows affected.


[('ANdrE WoOds', 78, 'Female', 'A+', 'Arthritis', datetime.datetime(2023, 2, 21, 0, 0), 'Sheila Rodgers', 'Carpenter Ltd', 'Blue Cross', Decimal('50089.29'), 320, 'Urgent', datetime.datetime(2023, 3, 6, 0, 0), 'Penicillin', 'Inconclusive'),
 ('ANdrE WoOds', 78, 'Female', 'A+', 'Arthritis', datetime.datetime(2023, 2, 21, 0, 0), 'Sheila Rodgers', 'Carpenter Ltd', 'Blue Cross', Decimal('50089.29'), 320, 'Urgent', datetime.datetime(2023, 3, 6, 0, 0), 'Penicillin', 'Inconclusive')]

From above 2 samples, it is evident 534 rows are duplicates.

## Instead of deleting duplicates from the table, I opted to create a VIEW to analyze the data.

In [181]:
%%sql
# Creating a 'View' Table

CREATE VIEW health_data as (WITH duplicates as (SELECT *, 
ROW_NUMBER() OVER (
    PARTITION BY name, age, doctor
    ORDER BY name
) as row_num
FROM healthcare_dataset)
SELECT name, age, gender,blood_type,medical_condition,
    date_of_admission, doctor, hospital, insurance_provider,
    billing_amount, room_number, admission_type,discharge_date,
    medication, test_results
FROM duplicates
    WHERE row_num = 1);

 * mysql+pymysql://root:***@localhost:3306/healthcare
0 rows affected.


[]

In [182]:
%%sql 
# Data from VIEW table

SELECT *
FROM health_data
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3306/healthcare
10 rows affected.


[('aAroN ADaMS', 38, 'Female', 'O-', 'Cancer', datetime.datetime(2021, 3, 10, 0, 0), 'Norma Li', 'Hart LLC', 'UnitedHealthcare', Decimal('26052.11'), 363, 'Elective', datetime.datetime(2021, 3, 29, 0, 0), 'Lipitor', 'Inconclusive'),
 ('aAROn aguIRRe', 36, 'Male', 'A-', 'Diabetes', datetime.datetime(2023, 2, 26, 0, 0), 'Katrina Luna', 'Murray-Shelton', 'UnitedHealthcare', Decimal('27087.56'), 300, 'Emergency', datetime.datetime(2023, 3, 13, 0, 0), 'Aspirin', 'Inconclusive'),
 ('AArOn AnderSoN', 50, 'Female', 'A+', 'Asthma', datetime.datetime(2020, 12, 18, 0, 0), 'Kenneth Jennings', 'Tanner-Cox', 'Cigna', Decimal('39804.66'), 196, 'Urgent', datetime.datetime(2021, 1, 15, 0, 0), 'Aspirin', 'Inconclusive'),
 ('aaRON AndeRSoN md', 20, 'Female', 'A-', 'Hypertension', datetime.datetime(2021, 3, 28, 0, 0), 'Tammy Perez', 'Ritter LLC', 'UnitedHealthcare', Decimal('16846.42'), 249, 'Elective', datetime.datetime(2021, 4, 9, 0, 0), 'Paracetamol', 'Abnormal'),
 ('AAron ArCHER', 47, 'Female', 'B-', 'Cancer', datetime.datetime(2021, 1, 10, 0, 0), 'Cynthia Villanueva', 'Montes Case and Mendez,', 'Medicare', Decimal('10602.08'), 108, 'Urgent', datetime.datetime(2021, 1, 17, 0, 0), 'Paracetamol', 'Inconclusive'),
 ('AAron ArCHER', 49, 'Female', 'B-', 'Cancer', datetime.datetime(2021, 1, 10, 0, 0), 'Cynthia Villanueva', 'Montes Case and Mendez,', 'Medicare', Decimal('10602.08'), 108, 'Urgent', datetime.datetime(2021, 1, 17, 0, 0), 'Paracetamol', 'Inconclusive'),
 ('AAroN bAker', 73, 'Male', 'B+', 'Cancer', datetime.datetime(2019, 6, 18, 0, 0), 'Tracy Torres', 'Wise and Todd, Parker', 'Medicare', Decimal('10135.89'), 234, 'Elective', datetime.datetime(2019, 7, 10, 0, 0), 'Aspirin', 'Inconclusive'),
 ('AaROn bAkEr', 84, 'Female', 'A+', 'Asthma', datetime.datetime(2022, 6, 4, 0, 0), 'Brittany Smith', 'Carter, Abbott and Fuentes', 'Medicare', Decimal('6826.68'), 496, 'Emergency', datetime.datetime(2022, 6, 10, 0, 0), 'Lipitor', 'Normal'),
 ('AARON bAldWIN Jr.', 20, 'Male', 'O-', 'Hypertension', datetime.datetime(2020, 10, 10, 0, 0), 'Amy Farley', 'Flores Friedman and White,', 'Medicare', Decimal('29740.96'), 104, 'Urgent', datetime.datetime(2020, 11, 5, 0, 0), 'Paracetamol', 'Abnormal'),
 ('aaROn bARNes', 85, 'Male', 'O-', 'Obesity', datetime.datetime(2023, 3, 30, 0, 0), 'Vincent Johnson', 'Taylor LLC', 'UnitedHealthcare', Decimal('4783.44'), 136, 'Urgent', datetime.datetime(2023, 4, 20, 0, 0), 'Ibuprofen', 'Normal')]

In [327]:
%%sql 
# Number of rows in the VIEW table

SELECT 
    COUNT(*) as dataset_row
FROM health_data;

 * mysql+pymysql://root:***@localhost:3306/healthcare
1 rows affected.


[(54966,)]

From the above observation, the VIEW lacks duplicates. Initially, the original table had 55,500 rows. Now, the VIEW has 54,966 rows, indicating the presence of 534 duplicate entries in the original table.

# Analysis

In [330]:
%%sql
# Monthly Patient Admissions for Each Year

SELECT  
    YEAR(date_of_admission) as year,
    MONTHNAME(date_of_admission) as month,
    count(*) admission_per_month
FROM health_data
GROUP BY 
    year,
    month
ORDER BY year DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
61 rows affected.


[(2024, 'April', 941),
 (2024, 'February', 869),
 (2024, 'January', 903),
 (2024, 'March', 902),
 (2024, 'May', 212),
 (2023, 'April', 893),
 (2023, 'August', 962),
 (2023, 'December', 916),
 (2023, 'February', 873),
 (2023, 'January', 926),
 (2023, 'July', 913),
 (2023, 'June', 953),
 (2023, 'March', 912),
 (2023, 'May', 918),
 (2023, 'November', 873),
 (2023, 'October', 884),
 (2023, 'September', 913),
 (2022, 'April', 871),
 (2022, 'August', 948),
 (2022, 'December', 953),
 (2022, 'February', 772),
 (2022, 'January', 960),
 (2022, 'July', 935),
 (2022, 'June', 959),
 (2022, 'March', 931),
 (2022, 'May', 885),
 (2022, 'November', 886),
 (2022, 'October', 903),
 (2022, 'September', 912),
 (2021, 'April', 862),
 (2021, 'August', 887),
 (2021, 'December', 944),
 (2021, 'February', 828),
 (2021, 'January', 924),
 (2021, 'July', 966),
 (2021, 'June', 915),
 (2021, 'March', 953),
 (2021, 'May', 893),
 (2021, 'November', 899),
 (2021, 'October', 885),
 (2021, 'September', 860),
 (2020, 'April', 911),
 (2020, 'August', 1003),
 (2020, 'December', 883),
 (2020, 'February', 868),
 (2020, 'January', 942),
 (2020, 'July', 1000),
 (2020, 'June', 924),
 (2020, 'March', 924),
 (2020, 'May', 970),
 (2020, 'November', 900),
 (2020, 'October', 948),
 (2020, 'September', 899),
 (2019, 'August', 985),
 (2019, 'December', 921),
 (2019, 'July', 951),
 (2019, 'June', 899),
 (2019, 'May', 677),
 (2019, 'November', 950),
 (2019, 'October', 993),
 (2019, 'September', 924)]

In [332]:
%%sql
# Monthly Patient Discharges per Year

SELECT 
    YEAR(discharge_date) as year,
    MONTHNAME(discharge_date) as month,
    COUNT(*) patients_discharged
FROM health_data
GROUP BY 
    year,
    month
ORDER BY year DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
62 rows affected.


[(2024, 'April', 871),
 (2024, 'February', 898),
 (2024, 'January', 872),
 (2024, 'June', 22),
 (2024, 'March', 896),
 (2024, 'May', 713),
 (2023, 'April', 893),
 (2023, 'August', 929),
 (2023, 'December', 915),
 (2023, 'February', 868),
 (2023, 'January', 937),
 (2023, 'July', 932),
 (2023, 'June', 952),
 (2023, 'March', 924),
 (2023, 'May', 914),
 (2023, 'November', 853),
 (2023, 'October', 946),
 (2023, 'September', 906),
 (2022, 'April', 908),
 (2022, 'August', 908),
 (2022, 'December', 906),
 (2022, 'February', 842),
 (2022, 'January', 961),
 (2022, 'July', 969),
 (2022, 'June', 889),
 (2022, 'March', 884),
 (2022, 'May', 889),
 (2022, 'November', 868),
 (2022, 'October', 978),
 (2022, 'September', 916),
 (2021, 'April', 858),
 (2021, 'August', 909),
 (2021, 'December', 955),
 (2021, 'February', 805),
 (2021, 'January', 922),
 (2021, 'July', 975),
 (2021, 'June', 875),
 (2021, 'March', 957),
 (2021, 'May', 924),
 (2021, 'November', 839),
 (2021, 'October', 859),
 (2021, 'September', 899),
 (2020, 'April', 919),
 (2020, 'August', 1022),
 (2020, 'December', 909),
 (2020, 'February', 875),
 (2020, 'January', 938),
 (2020, 'July', 962),
 (2020, 'June', 945),
 (2020, 'March', 911),
 (2020, 'May', 941),
 (2020, 'November', 890),
 (2020, 'October', 915),
 (2020, 'September', 965),
 (2019, 'August', 953),
 (2019, 'December', 924),
 (2019, 'July', 908),
 (2019, 'June', 843),
 (2019, 'May', 273),
 (2019, 'November', 1034),
 (2019, 'October', 912),
 (2019, 'September', 991)]

In [333]:
%%sql 
# Number of Patients by Gender for Each Medical Condition

SELECT 
    medical_condition,
    gender,
COUNT(*)as patients
FROM health_data
GROUP BY 
    medical_condition,
    gender
ORDER BY medical_condition;

 * mysql+pymysql://root:***@localhost:3306/healthcare
12 rows affected.


[('Arthritis', 'Female', 4642),
 ('Arthritis', 'Male', 4576),
 ('Asthma', 'Female', 4511),
 ('Asthma', 'Male', 4584),
 ('Cancer', 'Female', 4566),
 ('Cancer', 'Male', 4574),
 ('Diabetes', 'Female', 4609),
 ('Diabetes', 'Male', 4607),
 ('Hypertension', 'Female', 4569),
 ('Hypertension', 'Male', 4582),
 ('Obesity', 'Female', 4573),
 ('Obesity', 'Male', 4573)]

In [339]:
%%sql 
# Age MIN, MAX, and Average

SELECT 
    MIN(age) as lowest_age, 
    MAX(age) as highest_age,
    ROUND(AVG(age)) as average_age
FROM health_data;

 * mysql+pymysql://root:***@localhost:3306/healthcare
1 rows affected.


[(13, 89, Decimal('52'))]

In [342]:
%%sql
# Age Distrubution over specified range

SELECT 
    CASE 
        WHEN age BETWEEN 0 AND 9 THEN '0-9'
        WHEN age BETWEEN 10 AND 19 THEN '10-19'
        WHEN age BETWEEN 20 AND 29 THEN '20-29'
        WHEN age BETWEEN 30 AND 39 THEN '30-39'
        WHEN age BETWEEN 40 AND 49 THEN '40-49'
        WHEN age BETWEEN 50 AND 59 THEN '50-59'
        WHEN age BETWEEN 60 AND 69 THEN '60-69'
        WHEN age BETWEEN 70 AND 79 THEN '70-79'
        WHEN age BETWEEN 80 AND 89 THEN '80-89'
        ELSE '90+'
    END AS age_range,
    COUNT(*) AS age_count
FROM health_data
GROUP BY age_range
ORDER BY age_range;


 * mysql+pymysql://root:***@localhost:3306/healthcare
8 rows affected.


[('10-19', 1677),
 ('20-29', 7928),
 ('30-39', 8105),
 ('40-49', 8045),
 ('50-59', 8263),
 ('60-69', 8099),
 ('70-79', 7999),
 ('80-89', 4850)]

In [354]:
%%sql 
# Details of all youngest patients (aged 13)

SELECT 
    name,age,gender,blood_type,
    medical_condition, date_of_admission,
    discharge_date, hospital,admission_type,
    medication,test_results,billing_amount
FROM health_data
WHERE age = 13
ORDER BY date_of_admission DESC;


 * mysql+pymysql://root:***@localhost:3306/healthcare
14 rows affected.


[('rOnalD daVis', 13, 'Male', 'A+', 'Obesity', datetime.datetime(2023, 12, 6, 0, 0), datetime.datetime(2024, 1, 4, 0, 0), 'Kelly, and Gomez Williams', 'Emergency', 'Paracetamol', 'Normal', Decimal('3014.57')),
 ('cAthY BARNES', 13, 'Male', 'O-', 'Obesity', datetime.datetime(2023, 3, 30, 0, 0), datetime.datetime(2023, 4, 29, 0, 0), 'Wood-Johnson', 'Emergency', 'Lipitor', 'Abnormal', Decimal('29170.62')),
 ('dAWn chAveZ', 13, 'Male', 'AB+', 'Cancer', datetime.datetime(2023, 3, 13, 0, 0), datetime.datetime(2023, 3, 25, 0, 0), 'Taylor, Boyle Dalton and', 'Emergency', 'Ibuprofen', 'Abnormal', Decimal('43935.23')),
 ('CHArlEs joRDAN', 13, 'Male', 'A+', 'Arthritis', datetime.datetime(2021, 5, 13, 0, 0), datetime.datetime(2021, 6, 12, 0, 0), 'Arellano and Armstrong Hensley,', 'Emergency', 'Penicillin', 'Normal', Decimal('10597.38')),
 ('eLIzABETh cAsTILlO', 13, 'Male', 'AB-', 'Hypertension', datetime.datetime(2021, 1, 3, 0, 0), datetime.datetime(2021, 1, 27, 0, 0), 'Thomas-Pierce', 'Elective', 'Aspirin', 'Abnormal', Decimal('32900.04')),
 ('jamES BasS phD', 13, 'Male', 'O+', 'Asthma', datetime.datetime(2020, 12, 30, 0, 0), datetime.datetime(2021, 1, 17, 0, 0), 'Pollard Wallace, Sims and', 'Emergency', 'Ibuprofen', 'Inconclusive', Decimal('22901.02')),
 ('DEaNnA pALMeR', 13, 'Male', 'AB-', 'Obesity', datetime.datetime(2020, 9, 20, 0, 0), datetime.datetime(2020, 9, 23, 0, 0), 'and Sanchez Phillips, Brown', 'Emergency', 'Penicillin', 'Inconclusive', Decimal('23941.76')),
 ('doRothY hoffMAn', 13, 'Female', 'O-', 'Cancer', datetime.datetime(2020, 7, 18, 0, 0), datetime.datetime(2020, 8, 10, 0, 0), 'Davis Davis, and Davis', 'Elective', 'Ibuprofen', 'Abnormal', Decimal('22316.17')),
 ('JessICA CHuNG', 13, 'Male', 'AB+', 'Obesity', datetime.datetime(2020, 7, 6, 0, 0), datetime.datetime(2020, 7, 11, 0, 0), 'Davis-Lawrence', 'Urgent', 'Aspirin', 'Abnormal', Decimal('32019.91')),
 ('kYLE jAcKsoN', 13, 'Female', 'B-', 'Asthma', datetime.datetime(2020, 5, 24, 0, 0), datetime.datetime(2020, 6, 15, 0, 0), 'Small-Shaw', 'Emergency', 'Paracetamol', 'Abnormal', Decimal('31988.67')),
 ('JeNnIFer JoHNsoN', 13, 'Female', 'AB+', 'Arthritis', datetime.datetime(2020, 5, 3, 0, 0), datetime.datetime(2020, 5, 29, 0, 0), 'Banks Gordon and Browning,', 'Emergency', 'Aspirin', 'Abnormal', Decimal('30506.94')),
 ('kRiStiN orTIZ', 13, 'Male', 'A-', 'Asthma', datetime.datetime(2020, 4, 1, 0, 0), datetime.datetime(2020, 4, 2, 0, 0), 'Yu and Mckenzie Lucas,', 'Urgent', 'Paracetamol', 'Abnormal', Decimal('50526.71')),
 ('jAcob wILLiAMs', 13, 'Female', 'A+', 'Obesity', datetime.datetime(2020, 3, 3, 0, 0), datetime.datetime(2020, 3, 20, 0, 0), 'and Huerta, Cox Price', 'Urgent', 'Ibuprofen', 'Normal', Decimal('42349.11')),
 ('JoNaTHAn JacksoN', 13, 'Female', 'B-', 'Obesity', datetime.datetime(2020, 2, 7, 0, 0), datetime.datetime(2020, 2, 24, 0, 0), 'Spencer-Shields', 'Urgent', 'Aspirin', 'Inconclusive', Decimal('30075.23'))]

In [355]:
%%sql 
# Details of all oldest patients (aged 89)

SELECT 
    name,gender,blood_type,
    medical_condition,date_of_admission,
    discharge_date,
    hospital,admission_type,
    medication,test_results,billing_amount
FROM health_data
WHERE age = 89
ORDER BY date_of_admission DESC;


 * mysql+pymysql://root:***@localhost:3306/healthcare
8 rows affected.


[('mIchaeL POtTs', 'Male', 'A+', 'Hypertension', datetime.datetime(2023, 1, 19, 0, 0), datetime.datetime(2023, 1, 28, 0, 0), 'and Perez, Bennett Townsend', 'Emergency', 'Paracetamol', 'Abnormal', Decimal('9542.74')),
 ('doNALD aViLA', 'Female', 'AB-', 'Asthma', datetime.datetime(2022, 9, 17, 0, 0), datetime.datetime(2022, 9, 27, 0, 0), 'Holmes and Howard Castro,', 'Emergency', 'Aspirin', 'Abnormal', Decimal('14042.75')),
 ('hEAthEr dawsOn', 'Male', 'O-', 'Arthritis', datetime.datetime(2022, 7, 18, 0, 0), datetime.datetime(2022, 8, 13, 0, 0), 'Moore-Gray', 'Elective', 'Ibuprofen', 'Normal', Decimal('17172.12')),
 ('THomAs PHIllIpS', 'Female', 'A+', 'Hypertension', datetime.datetime(2022, 5, 22, 0, 0), datetime.datetime(2022, 5, 26, 0, 0), 'and Martin, Davidson Cox', 'Urgent', 'Penicillin', 'Inconclusive', Decimal('48274.35')),
 ('MiChAEL DOmINGuEz', 'Male', 'O+', 'Cancer', datetime.datetime(2021, 9, 7, 0, 0), datetime.datetime(2021, 9, 24, 0, 0), 'PLC White', 'Elective', 'Aspirin', 'Abnormal', Decimal('7628.95')),
 ('DeBORah McBrIdE', 'Female', 'O+', 'Obesity', datetime.datetime(2021, 5, 28, 0, 0), datetime.datetime(2021, 6, 13, 0, 0), 'and Miller, Olson Olson', 'Elective', 'Paracetamol', 'Abnormal', Decimal('5692.91')),
 ('DAVId NeWTOn', 'Female', 'O+', 'Arthritis', datetime.datetime(2021, 2, 8, 0, 0), datetime.datetime(2021, 2, 20, 0, 0), 'Cooper Brown Parks, and', 'Elective', 'Penicillin', 'Inconclusive', Decimal('34500.02')),
 ('JerEmY hArdIN JR.', 'Male', 'A+', 'Diabetes', datetime.datetime(2019, 11, 5, 0, 0), datetime.datetime(2019, 11, 28, 0, 0), 'Gray-Solomon', 'Elective', 'Penicillin', 'Normal', Decimal('7242.64'))]

In [357]:
%%sql
# Number of Patients per Test Result

SELECT 
    test_results,
    COUNT(*) test_results_values
FROM health_data
GROUP BY 
    test_results
ORDER BY test_results_values DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
3 rows affected.


[('Abnormal', 18437), ('Normal', 18331), ('Inconclusive', 18198)]

In [358]:
%%sql
# Number of Patients by Blood Type

SELECT 
    blood_type,
    COUNT(*) blood_type_values
FROM health_data
GROUP BY 
    blood_type
ORDER BY blood_type_values DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
8 rows affected.


[('A-', 6898),
 ('A+', 6896),
 ('B+', 6885),
 ('AB+', 6882),
 ('AB-', 6874),
 ('B-', 6872),
 ('O+', 6855),
 ('O-', 6804)]

In [359]:
%%sql
# Most Prescribed Drugs Ordered from Highest to Lowest

SELECT 
    medication, 
    COUNT(medication) prescri_drugs
FROM health_data
GROUP BY 
    medication
ORDER BY prescri_drugs DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
5 rows affected.


[('Lipitor', 11038),
 ('Ibuprofen', 11023),
 ('Aspirin', 10984),
 ('Paracetamol', 10965),
 ('Penicillin', 10956)]

In [360]:
%%sql
# Average Billing Amount per Hospital - Displaying the first 10 hospitals

SELECT 
    hospital, 
    ROUND(AVG(billing_amount),2) as avg_billing_amount
FROM health_data
GROUP BY 
    hospital
ORDER BY avg_billing_amount DESC
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3306/healthcare
10 rows affected.


[('Hernandez-Morton', Decimal('52373.03')),
 ('Walker-Garcia', Decimal('52170.04')),
 ('Ruiz-Anthony', Decimal('52154.24')),
 ('George-Gonzalez', Decimal('52102.24')),
 ('Rocha-Carter', Decimal('52092.67')),
 ('Briggs Walker Martinez, and', Decimal('52024.73')),
 ('and Small Stephens Harrington,', Decimal('51975.97')),
 ('Clark-Espinoza', Decimal('51848.20')),
 ('Pierce and Miller James,', Decimal('51722.12')),
 ('Stephens Ltd', Decimal('51714.30'))]

In [362]:
%%sql
# Number of Patients Covered by Each Insurance Provider

SELECT 
    insurance_provider, 
    COUNT(*) as patients
FROM health_data
GROUP BY 
    insurance_provider
ORDER BY patients DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
5 rows affected.


[('Cigna', 11139),
 ('Medicare', 11039),
 ('UnitedHealthcare', 11014),
 ('Blue Cross', 10952),
 ('Aetna', 10822)]

In [363]:
%%sql
# Number of Patients vs Admission Type

SELECT 
    admission_type,
    COUNT(*) adm_type_values
FROM health_data
GROUP BY 
    admission_type
ORDER BY adm_type_values DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
3 rows affected.


[('Elective', 18473), ('Urgent', 18391), ('Emergency', 18102)]

In [365]:
%%sql
# The number of patients varies by insurance provider and admission type.

SELECT 
    admission_type,
    insurance_provider,
    COUNT(*) patient_count
FROM health_data
GROUP BY 
    admission_type,
    insurance_provider
ORDER BY admission_type DESC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
15 rows affected.


[('Urgent', 'Aetna', 3463),
 ('Urgent', 'Blue Cross', 3646),
 ('Urgent', 'Cigna', 3825),
 ('Urgent', 'Medicare', 3759),
 ('Urgent', 'UnitedHealthcare', 3698),
 ('Emergency', 'Aetna', 3641),
 ('Emergency', 'Blue Cross', 3597),
 ('Emergency', 'Cigna', 3638),
 ('Emergency', 'Medicare', 3643),
 ('Emergency', 'UnitedHealthcare', 3583),
 ('Elective', 'Aetna', 3718),
 ('Elective', 'Blue Cross', 3709),
 ('Elective', 'Cigna', 3676),
 ('Elective', 'Medicare', 3637),
 ('Elective', 'UnitedHealthcare', 3733)]

In [31]:
%%sql

# Average Billing per Medical Condition category per gender

SELECT 
    medical_condition,
    gender, 
    ROUND(avg(billing_amount),2) as average_billing_amount
FROM health_data
GROUP BY 
    medical_condition,
    gender
ORDER BY medical_condition ASC;

 * mysql+pymysql://root:***@localhost:3306/healthcare
12 rows affected.


[('Arthritis', 'Female', Decimal('25337.23')),
 ('Arthritis', 'Male', Decimal('25688.85')),
 ('Asthma', 'Female', Decimal('25358.63')),
 ('Asthma', 'Male', Decimal('25903.91')),
 ('Cancer', 'Female', Decimal('25284.31')),
 ('Cancer', 'Male', Decimal('25020.56')),
 ('Diabetes', 'Female', Decimal('25558.06')),
 ('Diabetes', 'Male', Decimal('25762.94')),
 ('Hypertension', 'Female', Decimal('25434.79')),
 ('Hypertension', 'Male', Decimal('25571.14')),
 ('Obesity', 'Female', Decimal('25871.07')),
 ('Obesity', 'Male', Decimal('25737.65'))]

In [56]:
%%sql
# Calculating patient admission days

SELECT 
    name,
    age,
    gender,
    medical_condition,
    hospital,
    room_number,
    admission_type,
    DATE(date_of_admission) as admission_date,
    DATE(discharge_date) as discharged_date,
    DATEDIFF(discharge_date,date_of_admission) as days_elapsed
FROM health_data
ORDER BY date_of_admission
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3306/healthcare
10 rows affected.


[('TOnya sHAh', 52, 'Female', 'Obesity', 'Moss-Mathews', 169, 'Emergency', datetime.date(2019, 5, 8), datetime.date(2019, 5, 13), 5),
 ('sTEphen GAlLOwAy', 82, 'Male', 'Cancer', 'Group Robbins', 217, 'Emergency', datetime.date(2019, 5, 8), datetime.date(2019, 6, 1), 24),
 ('STePHen BoWen', 72, 'Male', 'Cancer', 'Bailey-Freeman', 217, 'Elective', datetime.date(2019, 5, 8), datetime.date(2019, 5, 25), 17),
 ('tArA KELlY', 47, 'Male', 'Cancer', 'and Sons Santos', 110, 'Elective', datetime.date(2019, 5, 8), datetime.date(2019, 5, 21), 13),
 ('RObeRt thOmPsoN', 67, 'Female', 'Cancer', 'Macias and Cardenas Oliver,', 194, 'Emergency', datetime.date(2019, 5, 8), datetime.date(2019, 6, 3), 26),
 ('scoTt phIllIpS', 25, 'Male', 'Diabetes', 'Kane Johnson, and Mayer', 127, 'Emergency', datetime.date(2019, 5, 8), datetime.date(2019, 5, 20), 12),
 ('TimOTHy sHephERD PHd', 54, 'Female', 'Cancer', 'and Phelps Miller Buchanan,', 101, 'Elective', datetime.date(2019, 5, 8), datetime.date(2019, 5, 27), 19),
 ('RegInA ZuNIgA', 29, 'Male', 'Obesity', 'Inc Castillo', 167, 'Emergency', datetime.date(2019, 5, 8), datetime.date(2019, 5, 31), 23),
 ('RegInA ZuNIgA', 25, 'Male', 'Obesity', 'Inc Castillo', 167, 'Emergency', datetime.date(2019, 5, 8), datetime.date(2019, 5, 31), 23),
 ('MoRGAn LAnE', 68, 'Male', 'Hypertension', 'Sanford-Nelson', 382, 'Elective', datetime.date(2019, 5, 8), datetime.date(2019, 5, 10), 2)]